### PR TITLE
Add getSeats() method to API

### PR DESCRIPTION
**Pull request by keithradford**
Pretty simple implementation. I am considering adding the parameter `term` to the method to increase usability, but we can discuss that since the original idea was just having `crn`.

In terms of the tests, I figured the actual scraper methods invoked by the API do enough in terms of testing actual data, so I'm just verifying the objects contain the expected properties. Plus, the API is making actual calls to UVic's website which would mean we have to test dynamic data,

### CODE REVIEW

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -1,17 +1,30 @@
 import { UvicCourseScraper } from '..';
 
-describe('client', () => {
-  it('works properly', async () => {
+describe('a random index from getAllCourses()', () => {
```
Comment (isaaccormack) [inline src/__tests__/index.test.ts:3]: maybe the following would read better? 
```
describe('call getAllCourses()')
    it('should have all required attributes for a random course')
    ...
```

note the distinction between _required_ and _expected_ as the data is required to have these attributes always, not just expected in this test case

File: src/__tests__/index.test.ts [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
+    expect(allCourses[courseIdx]).toHaveProperty('getDetails')
+    expect(allCourses[courseIdx]).toHaveProperty('id')
+    expect(allCourses[courseIdx]).toHaveProperty('pid')
+    expect(allCourses[courseIdx]).toHaveProperty('subjectCode')
+    expect(allCourses[courseIdx]).toHaveProperty('title')
+  });
+});
+
+describe('when getSeats() is envoked', () => {
+  it('has the expected data', async () => {
```
Comment (isaaccormack) [inline src/__tests__/index.test.ts:24]: same goes for here

File: src/index.ts [hunk 3]
```diff
@@ -4,10 +4,10 @@ import got from 'got';
 import { classScheduleListingUrl, detailedClassInformationUrl } from './common/urls';
 import { classScheduleListingExtractor } from './pages/courseListingEntries';
 import { detailedClassInfoExtractor } from './pages/detailedClassInformation';
-import { KualiCourseCatalog, KualiCourseItem } from './types';
-import { getCurrTerm as getCurrentTerm } from './utils';
+import { DetailedClassInformation, KualiCourseCatalog, KualiCourseItem } from './types';
+import { getCurrTerm as getCurrentTerm, getCurrTerm } from './utils';
```
Comment (isaaccormack) [inline src/index.ts:8]: whatever is going on here is ambiguous

File: src/utils.ts [hunk 4]
```diff
@@ -56,9 +56,9 @@ export function getCurrTerm(date: dayjs.Dayjs = dayjs()) {
   const currMonth = date.month();
   let month = '';
 
-  if (1 <= currMonth && currMonth < 5) {
+  if (0 <= currMonth && currMonth < 4) {
```
Comment (isaaccormack) [inline src/utils.ts:59]: was this broken initially or did the API change?

File: src/__tests__/index.test.ts [hunk 5]
```diff
@@ -1,17 +1,30 @@
 import { UvicCourseScraper } from '..';
 
-describe('client', () => {
-  it('works properly', async () => {
+describe('a random index from getAllCourses()', () => {
+  it('has all the expected attributes', async () => {
     const client = await UvicCourseScraper();
     const allCourses = await client.getAllCourses();
```
Comment (aomi) [inline src/__tests__/index.test.ts:6]: Do we want to mock the underlying request? I don't think our tests should be hitting UVic/Kuali

Comment (aomi) [reply src/__tests__/index.test.ts:6]: I heard `nock` is a decent mocking library for this perhaps "wink"

File: src/__tests__/index.test.ts [hunk 6]
```diff
@@ -1,17 +1,30 @@
...
+    expect(allCourses[courseIdx]).toHaveProperty('dateStart')
+    expect(allCourses[courseIdx]).toHaveProperty('getDetails')
+    expect(allCourses[courseIdx]).toHaveProperty('id')
+    expect(allCourses[courseIdx]).toHaveProperty('pid')
+    expect(allCourses[courseIdx]).toHaveProperty('subjectCode')
+    expect(allCourses[courseIdx]).toHaveProperty('title')
+  });
+});
+
+describe('when getSeats() is envoked', () => {
```
Comment (aomi) [inline src/__tests__/index.test.ts:23]: Isn't it "invoked"?

File: src/index.ts (hunk 3 above)
Comment (aomi) [reply src/index.ts:8]: ^ this seems a bit odd. I also agree on going with `getCurrentTerm`.

File: src/index.ts [hunk 7]
```diff
@@ -4,10 +4,10 @@ import got from 'got';
...
 
-const COURSES_URL = 'https://uvic.kuali.co/api/v1/catalog/courses/5d9ccc4eab7506001ae4c225';
+const COURSES_URL = 'https://uvic.kuali.co/api/v1/catalog/courses/5f21b66d95f09c001ac436a0';
```
Comment (aomi) [inline src/index.ts:10]: I feel we should store all of them and date them accordingly for now.  So like `COURSES_URL_F2020` or something.

File: src/index.ts [hunk 8]
```diff
@@ -74,7 +74,16 @@ export const UvicCourseScraper = async () => {
     }));
   };
 
+  const getSeats = async (crn: string): Promise<DetailedClassInformation> => {
+    const details = await fetchSectionDetails(crn, getCurrTerm());
+    return {
+      seats: details.seats,
+      waitListSeats: details.waitListSeats
+    }
```
Comment (aomi) [inline src/index.ts:82]:
```suggestion
    const { seats, waitlistSeats }= await fetchSectionDetails(crn, getCurrTerm());
    return { seats, waitListSeats }
```

File: src/utils.ts (hunk 4 above)
Comment (aomi) [reply src/utils.ts:59]: I don't think we ever had tests for this thinking about this now.

Comment (keithradford) [reply src/utils.ts:59]: I don't think it was used anywhere before

File: src/index.ts [hunk 9]
```diff
@@ -74,7 +80,19 @@ export const UvicCourseScraper = async () => {
...
+  /**
+   * Fetches the data on seats and waitlist seats for a given course section.
+   *
+   * @param crn ie. '12345', '20001'
+   * @return {Promise<DetailedClassInformation>}
+   */
+  const getSeats = async (crn: string): Promise<DetailedClassInformation> => {
+    const { seats, waitListSeats } = await fetchSectionDetails(crn, getCurrentTerm());
+    return { seats, waitListSeats };
+  };
```
Comment (aomi) [inline src/index.ts:92]: Did we figure out if CRNs are unique? ie can the same CRN be used between different semesters/years? If not we need to pass in a `term` as well.

File: src/index.ts [hunk 10]
```diff
@@ -4,10 +4,15 @@ import got from 'got';
...
+  DetailedClassInformation,
+  KualiCourseCatalog,
+  KualiCourseItem,
+  COURSES_URL_W2021 as COURSES_URL,
+} from './types';
+import { getCurrentTerm } from './utils';
 
-const COURSES_URL = 'https://uvic.kuali.co/api/v1/catalog/courses/5f21b66d95f09c001ac436a0';
+// TODO: Currently not in use with current implementations,
 const COURSE_DETAIL_URL = 'https://uvic.kuali.co/api/v1/catalog/course/5d9ccc4eab7506001ae4c225/';
```
Comment (aomi) [inline src/index.ts:16]: I think this can be removed since it's saved elsewhere.

File: src/__tests__/index.test.ts [hunk 11]
```diff
@@ -19,3 +31,25 @@ describe('a random index from getAllCourses()', () => {
     expect(allCourses[courseIdx]).toHaveProperty('title');
   });
 });
+
+describe('call getSeats()', () => {
+  it('has the expected data for a given section', async () => {
+    const htmlResponse = await getSectionFileByCRN('202101', '20001');
+    nock('https://www.uvic.ca')
+      .get('/BAN1P/bwckschd.p_disp_detail_sched?term_in=202101&crn_in=20001')
+      .reply(200, htmlResponse);
```
Comment (aomi) [inline src/__tests__/index.test.ts:40]: Heyyy this is slick.

**Review by aomi — APPROVED** [review]
Other than the one comment about CRNs LGTM.

File: src/index.ts (hunk 9 above)
Comment (isaaccormack) [reply src/index.ts:92]: same CRN can be used for different courses in different terms

**Review by isaaccormack — APPROVED** [review]

File: src/index.ts (hunk 9 above)
Comment (keithradford) [reply src/index.ts:92]: will add term parameter

**Pull request merged**